Artifacts/windows-7zip/ChocolateyPackageInstaller.ps1

```diff
@@ -0,0 +1,262 @@
+    WriteLog $("RawPackagesList : " + $RawPackagesList)
+    WriteLog "========== Configuration =========="
+}
+
+##################################################################################################
+
+# 
+# Description:
+#  - Creates the folder structure which'll be used for dumping logs generated by this script and
+#    the logon task.
+#
+# Parameters:
+#  - N/A.
+#
+# Return:
+#  - N/A.
+#
+# Notes:
+#  - N/A.
+#
+
+function InitializeFolders
+{
+    if ($false -eq (Test-Path -Path $ChocolateyPackageInstallerFolder))
+    {
+        New-Item -Path $ChocolateyPackageInstallerFolder -ItemType directory | Out-Null
+    }
+}
+
+##################################################################################################
+
+# 
+# Description:
+#  - Writes specified string to the console as well as to the script log (indicated by $ScriptLog).
+#
+# Parameters:
+#  - $message: The string to write.
+#
+# Return:
+#  - N/A.
+#
+# Notes:
+#  - N/A.
+#
+
+function WriteLog
+{
+    Param(
+        <# Can be null or empty #> $message
+    )
+
+    $timestampedMessage = $("[" + [System.DateTime]::Now + "] " + $message) | % {  
+        Write-Host -Object $_
+        Out-File -InputObject $_ -FilePath $ScriptLog -Append
+    }
+}
+
+##################################################################################################
+
+# 
+# Description:
+#  - Installs the chocolatey package manager.
+#
+# Parameters:
+#  - N/A.
+#
+# Return:
+#  - If installation is successful, then nothing is returned.
+#  - Else a detailed terminating error is thrown.
+#
+# Notes:
+#  - @TODO: Write to $chocolateyInstallLog log file.
+#  - @TODO: Currently no errors are being written to the log file ($chocolateyInstallLog). This needs to be fixed.
+#
+
+function InstallChocolatey
+{
+    Param(
+        [ValidateNotNullOrEmpty()] $chocolateyInstallLog
+    )
+
+    WriteLog "Installing Chocolatey..."
+
+    Invoke-Expression ((new-object net.webclient).DownloadString('https://chocolatey.org/install.ps1')) | Out-Null
+
+    WriteLog "Success."
+}
+
+##################################################################################################
+
+#
+# Description:
+#  - Installs the specified chocolatet packages on the machine.
+#
+# Parameters:
+#  - N/A.
+#
+# Return:
+#  - N/A.
+#
+# Notes:
+#  - N/A.
+#
+
+function InstallPackages
+{
+    Param(
+        [ValidateNotNullOrEmpty()][string] $packagesList
+    )
+
+    $Separator = @(";",",")
+    $SplitOption = [System.StringSplitOptions]::RemoveEmptyEntries
+    $packages = $packagesList.Trim().Split($Separator, $SplitOption)
+
+    if (0 -eq $packages.Count)
+    {
+        WriteLog $("No packages were specified. Exiting...")
+        return        
+    }
+
+    foreach ($package in $packages)
+    {
+        WriteLog $("Installing package: " + $package)
+
+        # install git via chocolatey
+        choco install $package --force --yes --acceptlicense --verbose | Out-Null 
```

Comment:
This installs the latest available version of specified package.We should probably consider the following: 
1. Using a LKG (last known good) version. 
2. Adding an optional parameter to this script to let user explicitly specify version of the chocolatey package.